### PR TITLE
MTTL 2496: Publish process closed event to record activity history

### PR DESCRIPTION
**Pull request by vkapenkin**
## Link to JIRA ticket

[JIRA ticket](https://hackney.atlassian.net/browse/MTTL-2496)

## Describe this PR

Raise ProcessClosed event on automated eligibility check failure

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace ProcessesApi.V1.Infrastructure.JWT
+{
+    public static class ProcessStoppedEventConstants
```
Comment (humulla) [inline ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs:3]: Can we rename this to ProcessClosedEventConstants as suggested in Jira

File: ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    {
+        public const string V1_VERSION = "v1";
+        public const string EVENTTYPE = "ProcessStoppedEvent";
```
Comment (humulla) [inline ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs:6]: Can we rename the eventType to `ProcessClosedEvent` as suggested in Jira

File: ProcessesApi.Tests/V1/E2ETests/Steps/UpdateSoleToJointProcessSteps.cs [hunk 3]
```diff
@@ -119,5 +128,32 @@ public async Task ThenTheProcessStateIsUpdatedToManualChecksFailed(UpdateProcess
             // Cleanup
             await _dbFixture.DynamoDbContext.DeleteAsync<ProcessesDb>(dbRecord.Id).ConfigureAwait(false);
         }
+
+        public async Task ThenTheProcessStoppedEventIsRaised(ISnsFixture snsFixture, Guid processId)
```
Comment (humulla) [inline ProcessesApi.Tests/V1/E2ETests/Steps/UpdateSoleToJointProcessSteps.cs:132]:
```suggestion
        public async Task ThenTheProcessClosedEventIsRaised(ISnsFixture snsFixture, Guid processId)
```

File: ProcessesApi.Tests/V1/UseCase/SoleToJointServiceTests.cs [hunk 4]
```diff
@@ -251,5 +261,32 @@ public async Task ProcessStateIsUpdatedToCloseProcess(string fromState)
                                                  triggerObject);
             process.PreviousStates.LastOrDefault().State.Should().Be(fromState);
         }
+
+        [Fact]
+        public async Task ProcessStoppedEventIsRaisedWhenAutomaticEligibilityChecksFail()
```
Comment (humulla) [inline ProcessesApi.Tests/V1/UseCase/SoleToJointServiceTests.cs:266]:
```suggestion
        public async Task ProcessClosedEventIsRaisedWhenAutomaticEligibilityChecksFail()
```

File: ProcessesApi/V1/Factories/ProcessesSnsFactory.cs [hunk 5]
```diff
@@ -27,5 +27,29 @@ public EntityEventSns ProcessStarted(Process process, Token token)
                 User = new User { Name = token.Name, Email = token.Email }
             };
         }
+
+        public EntityEventSns ProcessStopped(Process process, Token token, string description)
```
Comment (humulla) [inline ProcessesApi/V1/Factories/ProcessesSnsFactory.cs:31]:
```suggestion
        public EntityEventSns ProcessClosed(Process process, Token token, string description)
```

File: ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs (hunk 1 above)
Comment (vkapenkin) [reply ProcessesApi/V1/Infrastructure/JWT/ProcessStoppedEventConstants.cs:3]: oops, was pretty sure it's named ProcessStopped :) Will fix

**Review by humulla — APPROVED** [review]

**Pull request merged**